### PR TITLE
Warn if runtimeconfig.template.json is used with PublishAot

### LINKED ISSUE #72310: runtimeconfig.template support
We should either support it with NativeAOT or clearly error out asking people to use the MSBuild properties instead.

### PR DESCRIPTION
**Pull request by MichalStrehovsky**
Fixes #72310.

Cc @dotnet/ilc-contrib

### CODE REVIEW

File: src/coreclr/nativeaot/BuildIntegration/Microsoft.NETCore.Native.Publish.targets [hunk 1]
```diff
@@ -40,6 +40,8 @@
   <UsingTask TaskName="ComputeManagedAssembliesToCompileToNative" AssemblyFile="$(IlcBuildTasksPath)" />
   <Target Name="_ComputeAssembliesToCompileToNative" DependsOnTargets="$(IlcDynamicBuildPropertyDependencies)">
 
+    <Warning Condition="Exists($(UserRuntimeConfig))" Text="The published project has a runtimeconfig.template.json, but PublishAot doesn't support this configuration format. If possible, move the configuration to the project file using RuntimeHostConfigurationOption." />
```
Comment (jkotas) [inline src/coreclr/nativeaot/BuildIntegration/Microsoft.NETCore.Native.Publish.targets:43]:
```suggestion
    <Warning Condition="Exists($(UserRuntimeConfig))" Text="The published project has a runtimeconfig.template.json that is not supported by PublishAot. Move the configuration to the project file using RuntimeHostConfigurationOption." />
```
Are there common situations where it is not possible to use RuntimeHostConfigurationOption as a replacement?

**Review by jkotas — APPROVED** [review]

**Comment by MichalStrehovsky** [comment]
Relevant pipelines are green.

**Comment by carlossanlop** [comment]
@MichalStrehovsky @jkotas is it ok to have this change in 7.0? Since we had to delay the snap for a full day, this commit will end up getting included in the RC1 snap. Letting you know in case you want to do something about it.

**Comment by MichalStrehovsky** [comment]
> @MichalStrehovsky @jkotas is it ok to have this change in 7.0? Since we had to delay the snap for a full day, this commit will end up getting included in the RC1 snap. Letting you know in case you want to do something about it.

Yes, all my PRs from yesterday were taking advantage of the moved deadline. I've fixed several papercut-level issues that I'm happy to see fixed for 7.0.

**Pull request merged**